### ISSUE TITLE
Server command '--suppress-adb-kill-server' displays appium.js: error: Unrecognized arguments:

### ISSUE THREAD

**Issue by whopiecat** [appium/appium#4904]
Hello:

The Server command '--suppress-adb-kill-server' displays appium.js: error: Unrecognized arguments: --suppress-adb-kill-server.

I have tried:  --suppress-adb-kill-server
I have tried:  --suppress-adb-kill-server true
Both return: appium.js: error: Unrecognized arguments:

The server does not start so no log.

The expected syntax can be found here : https://github.com/appium/appium/blob/d8cbf6b31b3af0bc1a617d283b52a46c7a9d6b24/docs/en/writing-running-appium/server-args.md

I noticed that the "--suppress-adb-kill-server", command is not displayed in the documentation at Appium.io.

Please provide the proper syntax if the code is working.

Thanks

**Comment by jlipps** [comment]
what is the full command you're running to start appium? what version of appium are you using?

**Comment by whopiecat** [comment]
Hello:
"appium --version" returns 1.3.7 as my versiohn of Appium.
my command line on a MAC is : 

/Applications/Appium.app/Contents/Resources/node_modules/appium/bin/appium.js --address <tcpip> --bootstrap-port 4723 --no-reset --local-timezone --nodeconfig <pathToFile>nodeconfigSM-T210R.json -p 4723 -U <device tcpip> 

The line above starts the Appium server and registers the device with the existing Selenium Grid.
Test are running fine on real devices (browser only).

If I include the command "--suppress-adb-kill-server" or "--suppress-adb-kill-server true" the server does not start up.

**Comment by imurchie** [comment]
`suppress-adb-kill-server` is not available in 1.3.7. It is in master and should be in the next release (I assume. @0x1mason the [commit](https://github.com/jpkrause/appium/commit/d8cbf6b31b3af0bc1a617d283b52a46c7a9d6b24) for this functionality seems to be from a long time ago, but it didn't make it into the release).

**Comment by jlipps** [comment]
also, `appium` and `/Applications/Appium.app/Contents/Resources/node_modules/appium/bin/appium.js` are potentially different versions of appium.

**Comment by whopiecat** [comment]
Thanks for the update.
When is the next release expected?

Without the "--suppress-adb-kill-server", call the ability to run Android test in parallel on a Selenium Grid is not possible.
Each separate start of a test on a node will fire the "kill-server" call and drop all devices IP connections.

Thanks

**Comment by jlipps** [comment]
I also just added a change that ensures that flag is getting stored internally as a boolean rather than an array (which was a bug)

**Comment by moizjv** [comment]
@whopiecat it should be available with `1.4.0 beta`

**Comment by whopiecat** [comment]
Thanks for the info.

-------- Original message --------
Subject: Re: [appium] Server command '--suppress-adb-kill-server' displays appium.js: error: Unrecognized arguments: (#4904)

**Comment by moizjv** [comment]
Closing the issue. If you still have issue in `1.4.0` with this argument open a new ticket. Thanks
